### PR TITLE
Add DNSSETSERVER type for a lwip ioctl to call dns_setserver()

### PR DESCRIPTION
**Pull request by HONGCHAEHEE**
- Existing dns client app and dhcpc call dns_setserver() which is in the kernel space directly.
The new code is added to call it by the ioctl function.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [77ea692d66abe5a0ead0be553a083d3e9da9db4a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [77ea692d66abe5a0ead0be553a083d3e9da9db4a] - Code Rule Check OK.

File: apps/examples/dnsclient_test/dnsclient_main.c [hunk 1]
```diff
@@ -147,7 +148,27 @@ int dnsclient_main(int argc, FAR char *argv[])
...
+		}
+
+		memset(&req, 0, sizeof(req));
+		req.type = DNSSETSERVER;
+		req.num_dns = 0;
+		req.dns_server = &dns_addr;
+
+		int ret = ioctl(sockfd, SIOCLWIP, (unsigned long)&req);
+		if (ret == ERROR) {
+			printf("dnsclient : ioctl() failed with errno: %d\n", errno);
```
Comment (sunghan-chang) [inline apps/examples/dnsclient_test/dnsclient_main.c]: Let's include `errno.h` to use errno.

File: apps/examples/dnsclient_test/dnsclient_main.c [hunk 2]
```diff
@@ -147,7 +148,27 @@ int dnsclient_main(int argc, FAR char *argv[])
...
+			return -1;
+		}
+
+		memset(&req, 0, sizeof(req));
+		req.type = DNSSETSERVER;
+		req.num_dns = 0;
+		req.dns_server = &dns_addr;
+
+		int ret = ioctl(sockfd, SIOCLWIP, (unsigned long)&req);
+		if (ret == ERROR) {
```
Comment (sunghan-chang) [inline apps/examples/dnsclient_test/dnsclient_main.c]: include `types.h` for ERROR

**Comment by seinfra** [comment]
Target : [d3e379285e2d0860000c2696e48b695dc4f05606] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [d3e379285e2d0860000c2696e48b695dc4f05606] - Code Rule Check (C++) OK.

File: apps/examples/dnsclient_test/dnsclient_main.c (hunk 1 above)
Comment (HONGCHAEHEE) [reply apps/examples/dnsclient_test/dnsclient_main.c]: Ok thank you.

File: apps/examples/dnsclient_test/dnsclient_main.c (hunk 2 above)
Comment (HONGCHAEHEE) [reply apps/examples/dnsclient_test/dnsclient_main.c]: Ok thank you.

**Comment by seinfra** [comment]
Target : [4665a89f357bc21bea52d4f3fa6b4cec46a0207c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [4665a89f357bc21bea52d4f3fa6b4cec46a0207c] - Code Rule Check OK.

**Pull request merged**